test_zipp.py

```diff
@@ -416,3 +417,23 @@ def test_inheritance(self, alpharep):
         for alpharep in self.zipfile_alpharep():
             file = cls(alpharep).joinpath('some dir').parent
             assert isinstance(file, cls)
+
+    def test_can_pickle_string_path(self):
+        path_1 = zipp.Path("/path/to/a/file.zip")
```

Comment:
As I started experimenting with this change, I realized that this assertion is a major deviation from the current expectation. Previously, `Path(file_that_does_not_exist)` would raise a `FileNotFoundError` or similar. Now, that `FileNotFoundError` is only encountered later when one attempts to access the contents of the zipfile. That's probably undesirable or at least a backward incompatibility.

Reply:
In 6f700a439e7535878fdc5639e6368ee3a9691221, attempting to construct a `zipp.Path` with a non-existent path will fail the same as always. It's only possible to pickle a `Path` that was constructed with an extant path.